Incompatible version between Spline and Spark BigQuery connector

I encounterd an exception as following when I tried to send Spark's lineage message to Kafka cluster.
```
24/07/29 22:56:26 ERROR org.apache.spark.util.Utils: throw uncaught fatal error in thread spark-listener-group-shared

java.lang.ExceptionInInitializerError

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$relationProviderProcessor$1.bigQueryConfig$1(BigQueryPlugin.scala:68)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$relationProviderProcessor$1.$anonfun$applyOrElse$7(BigQueryPlugin.scala:98)

    at scala.Option.orElse(Option.scala:447)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$relationProviderProcessor$1.applyOrElse(BigQueryPlugin.scala:98)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$relationProviderProcessor$1.applyOrElse(BigQueryPlugin.scala:62)

    at scala.PartialFunction$OrElse.apply(PartialFunction.scala:172)

    at za.co.absa.spline.harvester.plugin.composite.SaveIntoDataSourceCommandPlugin$$anonfun$writeNodeProcessor$1.applyOrElse(SaveIntoDataSourceCommandPlugin.scala:49)

    at za.co.absa.spline.harvester.plugin.composite.SaveIntoDataSourceCommandPlugin$$anonfun$writeNodeProcessor$1.applyOrElse(SaveIntoDataSourceCommandPlugin.scala:45)

    at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)

    at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)

    at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)

    at scala.PartialFunction$OrElse.applyOrElse(PartialFunction.scala:176)

    at scala.PartialFunction$Lifted.apply(PartialFunction.scala:228)

    at scala.PartialFunction$Lifted.apply(PartialFunction.scala:224)

    at za.co.absa.spline.harvester.builder.write.PluggableWriteCommandExtractor.asWriteCommand(PluggableWriteCommandExtractor.scala:45)

    at za.co.absa.spline.harvester.LineageHarvester.$anonfun$tryExtractWriteCommand$1(LineageHarvester.scala:145)

    at scala.util.Try$.apply(Try.scala:213)

    at za.co.absa.spline.harvester.LineageHarvester.tryExtractWriteCommand(LineageHarvester.scala:145)

    at za.co.absa.spline.harvester.LineageHarvester.harvest(LineageHarvester.scala:61)

    at za.co.absa.spline.agent.SplineAgent$$anon$1.$anonfun$handle$1(SplineAgent.scala:91)

    at za.co.absa.spline.agent.SplineAgent$$anon$1.withErrorHandling(SplineAgent.scala:100)

    at za.co.absa.spline.agent.SplineAgent$$anon$1.handle(SplineAgent.scala:72)

    at za.co.absa.spline.harvester.listener.QueryExecutionListenerDelegate.onSuccess(QueryExecutionListenerDelegate.scala:28)

    at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener.$anonfun$onSuccess$1(SplineQueryExecutionListener.scala:41)

    at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener.$anonfun$onSuccess$1$adapted(SplineQueryExecutionListener.scala:41)

    at scala.Option.foreach(Option.scala:407)

    at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener.onSuccess(SplineQueryExecutionListener.scala:41)

    at org.apache.spark.sql.util.ExecutionListenerBus.doPostEvent(QueryExecutionListener.scala:155)

    at org.apache.spark.sql.util.ExecutionListenerBus.doPostEvent(QueryExecutionListener.scala:131)

    at org.apache.spark.util.ListenerBus.postToAll(ListenerBus.scala:117)

    at org.apache.spark.util.ListenerBus.postToAll$(ListenerBus.scala:101)

    at org.apache.spark.sql.util.ExecutionListenerBus.postToAll(QueryExecutionListener.scala:131)

    at org.apache.spark.sql.util.ExecutionListenerBus.onOtherEvent(QueryExecutionListener.scala:135)

    at org.apache.spark.scheduler.SparkListenerBus.doPostEvent(SparkListenerBus.scala:100)

    at org.apache.spark.scheduler.SparkListenerBus.doPostEvent$(SparkListenerBus.scala:28)

    at org.apache.spark.scheduler.AsyncEventQueue.doPostEvent(AsyncEventQueue.scala:37)

    at org.apache.spark.scheduler.AsyncEventQueue.doPostEvent(AsyncEventQueue.scala:37)

    at org.apache.spark.util.ListenerBus.postToAll(ListenerBus.scala:117)

    at org.apache.spark.util.ListenerBus.postToAll$(ListenerBus.scala:101)

    at org.apache.spark.scheduler.AsyncEventQueue.super$postToAll(AsyncEventQueue.scala:105)

    at org.apache.spark.scheduler.AsyncEventQueue.$anonfun$dispatch$1(AsyncEventQueue.scala:105)

    at scala.runtime.java8.JFunction0$mcJ$sp.apply(JFunction0$mcJ$sp.java:23)

    at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)

    at org.apache.spark.scheduler.AsyncEventQueue.org$apache$spark$scheduler$AsyncEventQueue$$dispatch(AsyncEventQueue.scala:100)

    at org.apache.spark.scheduler.AsyncEventQueue$$anon$2.$anonfun$run$1(AsyncEventQueue.scala:96)

    at org.apache.spark.util.Utils$.tryOrStopSparkContext(Utils.scala:1381)

    at org.apache.spark.scheduler.AsyncEventQueue$$anon$2.run(AsyncEventQueue.scala:96)

Caused by: java.lang.RuntimeException: Cannot find method `public static SparkBigQueryConfig from(... {7 args} ...)` in the class `class com.google.cloud.spark.bigquery.SparkBigQueryConfig`

    at scala.sys.package$.error(package.scala:30)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$SparkBigQueryConfig$.$anonfun$methodFrom$2(BigQueryPlugin.scala:150)

    at scala.Option.getOrElse(Option.scala:189)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$SparkBigQueryConfig$.<init>(BigQueryPlugin.scala:150)

    at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$SparkBigQueryConfig$.<clinit>(BigQueryPlugin.scala)

    ... 47 more
```

Then, I went to check the commits history records of [BigQueryPlugin.scala](https://github.com/AbsaOSS/spline-spark-agent/blob/develop/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala#L150). I found that there has always been a check for whether the number of parameter types equals 7, right after the initial commit.

Hence, base on the prompt, I go to SparkBigQueryConfig.java and check the commits history records as following:
[v0.25.0](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/f1f7b2ddd96ff3596034f68d7d2c0df6ee968f21/spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java#L180)
[v0.25.1](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/c6487a8e721c83e630b82fec53a4f373df23aa4f/spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java#L187)

Base on the above commits history, it indeed provides a static SparkBigQueryConfig with 7 args  before version 0.25.0, but it was changed after 0.25.0.
Therefore, you will encounter this error if you use the version after this version.

--- a/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
@@ -16,7 +16,12 @@
 
 package za.co.absa.spline.harvester.plugin.embedded
 
+import java.lang.reflect.Method
+import java.lang.reflect.Modifier.isStatic
+import java.util.Optional
+
 import io.github.classgraph.ClassGraph
+import javax.annotation.Priority
 import org.apache.hadoop.conf.Configuration
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.execution.datasources.{LogicalRelation, SaveIntoDataSourceCommand}
@@ -31,10 +36,6 @@ import za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin.SparkBigQueryC
 import za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin._
 import za.co.absa.spline.harvester.plugin.{BaseRelationProcessing, Plugin, RelationProviderProcessing}
 
-import java.lang.reflect.Method
-import java.lang.reflect.Modifier.isStatic
-import java.util.Optional
-import javax.annotation.Priority
 import scala.collection.JavaConverters._
 import scala.language.reflectiveCalls
 
@@ -142,14 +143,39 @@ object BigQueryPlugin {
       def getParentProjectId: String
     }
     private val clazz = findPossiblyShadedClass("com.google.cloud", "com.google.cloud.spark.bigquery.SparkBigQueryConfig")
-    private val methodFrom: Method = clazz
+    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
+      case (a, b, c, d, e, f, g, h, i) =>
+        if (methodFrom7.isDefined)
+          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
+        else if (methodFrom8.isDefined)
+          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+        else
+          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+    }
+    private val methodFrom7: Option[Method] = clazz
+      .getMethods
+      .find(
+        m => m.getName == "from"
+          && isStatic(m.getModifiers)
+          && m.getParameterTypes.length == 7
+          && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
+      )
+    private val methodFrom8: Option[Method] = clazz
+      .getMethods
+      .find(
+        m => m.getName == "from"
+          && isStatic(m.getModifiers)
+          && m.getParameterTypes.length == 8
+          && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
+      )
+    private val methodFrom9: Option[Method] = clazz
       .getMethods
       .find(
         m => m.getName == "from"
           && isStatic(m.getModifiers)
           && m.getParameterTypes.length == 9
           && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
-      ).getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {9 args} ...)` in the class `$clazz`"))
+      )
 
     object ImmutableMap {
       type ImmutableMap = AnyRef
@@ -166,10 +192,10 @@ object BigQueryPlugin {
           .asInstanceOf[ImmutableMap]
     }
 
-    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig =
-      methodFrom
-        .invoke(clazz, _, _, _, _, _, _, _, _, _)
-        .asInstanceOf[SparkBigQueryConfig]
+    private val methodFrom: Method = methodFrom7
+      .orElse(methodFrom8)
+      .orElse(methodFrom9)
+      .getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {7|8|9 args} ...)` in the class `$clazz`"))
   }
 
   private object `_: DirectBigQueryRelation` extends SafeTypeMatchingExtractor[AnyRef]("com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation")

--- a/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
@@ -167,15 +167,10 @@ object BigQueryPlugin {
           && m.getParameterTypes.length == 9
           && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
       )
-    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
-      case (a, b, c, d, e, f, g, h, i) =>
-        if (methodFrom7.isDefined)
-          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
-        else if (methodFrom8.isDefined)
-          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
-        else
-          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
-    }
+    private val methodFrom: Method = methodFrom7
+      .orElse(methodFrom8)
+      .orElse(methodFrom9)
+      .getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {7|8|9 args} ...)` in the class `$clazz`"))
 
     object ImmutableMap {
       type ImmutableMap = AnyRef
@@ -191,10 +186,16 @@ object BigQueryPlugin {
           .invoke(imClass)
           .asInstanceOf[ImmutableMap]
     }
-    private val methodFrom: Method = methodFrom7
-      .orElse(methodFrom8)
-      .orElse(methodFrom9)
-      .getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {7|8|9 args} ...)` in the class `$clazz`"))
+
+    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
+      case (a, b, c, d, e, f, g, h, i) =>
+        if (methodFrom7.isDefined)
+          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
+        else if (methodFrom8.isDefined)
+          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+        else
+          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+    }
   }
 
   private object `_: DirectBigQueryRelation` extends SafeTypeMatchingExtractor[AnyRef]("com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation")

--- a/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
@@ -16,12 +16,7 @@
 
 package za.co.absa.spline.harvester.plugin.embedded
 
-import java.lang.reflect.Method
-import java.lang.reflect.Modifier.isStatic
-import java.util.Optional
-
 import io.github.classgraph.ClassGraph
-import javax.annotation.Priority
 import org.apache.hadoop.conf.Configuration
 import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.execution.datasources.{LogicalRelation, SaveIntoDataSourceCommand}
@@ -36,6 +31,10 @@ import za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin.SparkBigQueryC
 import za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin._
 import za.co.absa.spline.harvester.plugin.{BaseRelationProcessing, Plugin, RelationProviderProcessing}
 
+import java.lang.reflect.Method
+import java.lang.reflect.Modifier.isStatic
+import java.util.Optional
+import javax.annotation.Priority
 import scala.collection.JavaConverters._
 import scala.language.reflectiveCalls
 
@@ -143,33 +142,14 @@ object BigQueryPlugin {
       def getParentProjectId: String
     }
     private val clazz = findPossiblyShadedClass("com.google.cloud", "com.google.cloud.spark.bigquery.SparkBigQueryConfig")
-    private val methodFrom7: Option[Method] = clazz
+    private val methodFrom: Method = clazz
       .getMethods
       .find(
         m => m.getName == "from"
           && isStatic(m.getModifiers)
-          && m.getParameterTypes.length == 7
+          && (m.getParameterTypes.length == 7 || m.getParameterTypes.length == 8 || m.getParameterTypes.length == 9)
           && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
       )
-    private val methodFrom8: Option[Method] = clazz
-      .getMethods
-      .find(
-        m => m.getName == "from"
-          && isStatic(m.getModifiers)
-          && m.getParameterTypes.length == 8
-          && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
-      )
-    private val methodFrom9: Option[Method] = clazz
-      .getMethods
-      .find(
-        m => m.getName == "from"
-          && isStatic(m.getModifiers)
-          && m.getParameterTypes.length == 9
-          && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
-      )
-    private val methodFrom: Method = methodFrom7
-      .orElse(methodFrom8)
-      .orElse(methodFrom9)
       .getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {7|8|9 args} ...)` in the class `$clazz`"))
 
     object ImmutableMap {
@@ -188,13 +168,13 @@ object BigQueryPlugin {
     }
 
     val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
-      case (a, b, c, d, e, f, g, h, i) =>
-        if (methodFrom7.isDefined)
-          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
-        else if (methodFrom8.isDefined)
-          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
-        else
-          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+      case (a, b, c, d, e, f, g, h, i) => {
+        methodFrom.getParameterTypes.length match {
+          case 7 => methodFrom.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
+          case 8 => methodFrom.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+          case 9 => methodFrom.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+        }
+      }
     }
   }
 

--- a/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
@@ -66,13 +66,15 @@ class BigQueryPlugin(spark: SparkSession)
 
       def bigQueryConfig = {
         SparkBigQueryConfig.from(
-          ImmutableMap.copyOf(cmd.options.asJava),
+          cmd.options.asJava,
           ImmutableMap.copyOf(spark.conf.getAll.asJava),
           spark.sparkContext.hadoopConfiguration,
+          ImmutableMap.of,
           0,
           spark.sessionState.conf,
           spark.version,
-          Optional.empty()
+          Optional.empty(),
+          true
         )
       }
 
@@ -145,9 +147,9 @@ object BigQueryPlugin {
       .find(
         m => m.getName == "from"
           && isStatic(m.getModifiers)
-          && m.getParameterTypes.length == 7
+          && m.getParameterTypes.length == 9
           && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
-      ).getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {7 args} ...)` in the class `$clazz`"))
+      ).getOrElse(sys.error(s"Cannot find method `public static SparkBigQueryConfig from(... {9 args} ...)` in the class `$clazz`"))
 
     object ImmutableMap {
       type ImmutableMap = AnyRef
@@ -157,11 +159,16 @@ object BigQueryPlugin {
           .getMethod("copyOf", classOf[java.util.Map[_, _]])
           .invoke(imClass, _)
           .asInstanceOf[ImmutableMap]
+      val of: ImmutableMap =
+        imClass
+          .getMethod("of")
+          .invoke(imClass)
+          .asInstanceOf[ImmutableMap]
     }
 
-    val from: (ImmutableMap.ImmutableMap, ImmutableMap.ImmutableMap, Configuration, Integer, SQLConf, String, Optional[StructType]) => SparkBigQueryConfig =
+    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig =
       methodFrom
-        .invoke(clazz, _, _, _, _, _, _, _)
+        .invoke(clazz, _, _, _, _, _, _, _, _, _)
         .asInstanceOf[SparkBigQueryConfig]
   }
 

--- a/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
+++ b/core/src/main/scala/za/co/absa/spline/harvester/plugin/embedded/BigQueryPlugin.scala
@@ -143,15 +143,6 @@ object BigQueryPlugin {
       def getParentProjectId: String
     }
     private val clazz = findPossiblyShadedClass("com.google.cloud", "com.google.cloud.spark.bigquery.SparkBigQueryConfig")
-    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
-      case (a, b, c, d, e, f, g, h, i) =>
-        if (methodFrom7.isDefined)
-          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
-        else if (methodFrom8.isDefined)
-          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
-        else
-          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
-    }
     private val methodFrom7: Option[Method] = clazz
       .getMethods
       .find(
@@ -176,6 +167,15 @@ object BigQueryPlugin {
           && m.getParameterTypes.length == 9
           && m.getReturnType.getSimpleName == "SparkBigQueryConfig"
       )
+    val from: (java.util.Map[_, _], ImmutableMap.ImmutableMap, Configuration, ImmutableMap.ImmutableMap, Integer, SQLConf, String, Optional[StructType], java.lang.Boolean) => SparkBigQueryConfig = {
+      case (a, b, c, d, e, f, g, h, i) =>
+        if (methodFrom7.isDefined)
+          methodFrom7.get.invoke(clazz, a, b, c, e, f, g, h).asInstanceOf[SparkBigQueryConfig]
+        else if (methodFrom8.isDefined)
+          methodFrom8.get.invoke(clazz, a, b, c, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+        else
+          methodFrom9.get.invoke(clazz, a, b, c, d, e, f, g, h, i).asInstanceOf[SparkBigQueryConfig]
+    }
 
     object ImmutableMap {
       type ImmutableMap = AnyRef
@@ -191,7 +191,6 @@ object BigQueryPlugin {
           .invoke(imClass)
           .asInstanceOf[ImmutableMap]
     }
-
     private val methodFrom: Method = methodFrom7
       .orElse(methodFrom8)
       .orElse(methodFrom9)

